ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -33,15 +33,25 @@ def behavior_spec_from_proto(
     :param agent_info: protobuf object.
     :return: BehaviorSpec object.
     """
-    observation_specs = []
-    for obs in agent_info.observations:
-        observation_specs.append(
-            ObservationSpec(
-                tuple(obs.shape),
-                tuple(DimensionProperty(dim) for dim in obs.dimension_properties),
-                ObservationType(obs.observation_type),
-            )
-        )
+    observation_shape = [tuple(obs.shape) for obs in agent_info.observations]
+    dim_props = [
+        tuple(DimensionProperty(dim) for dim in obs.dimension_properties)
+        for obs in agent_info.observations
+    ]
+    dim_props = [
+        dim_prop
+        if len(dim_prop) > 0
+        else (DimensionProperty.UNSPECIFIED,) * len(observation_shape[idx])
```

Comment:
I suppose it's too late to default this value in the communicator?

Reply:
rpc_utils is called inside the communicator. I need the default message to be empty so I can accommodate for past versions of the communicator.